### PR TITLE
Add option `g:vrc_enable_header_buffer` to filter curl header into a separate buffer

### PR DESCRIPTION
**Pull request by hiberabyss**

### CODE REVIEW

File: ftplugin/rest.vim [hunk 1]
```diff
@@ -659,6 +672,11 @@ function! s:DisplayOutput(tmpBufName, outputInfo, config)
       endif
     endif
 
+    if exists("g:vrc_enable_header_buffer") && g:vrc_enable_header_buffer
```
Comment (diepm) [inline ftplugin/rest.vim]: `if s:GetOpt('vrc_enable_header_buffer', 0)`

File: ftplugin/rest.vim [hunk 2]
```diff
@@ -565,6 +565,19 @@ function! s:GetCurlDataArgs(request)
...
+
+function! s:CreateHeaderBuf(headerStr)
+    execute(":e " . g:vrc_header_buffer_name)
+    setl buftype=nofile
+    execute(":%d")
+    call setline(1, split(a:headerStr, '\v\n'))
+    let responseBufName = s:GetOpt('vrc_output_buffer_name', '__REST_response__')
+    execute(printf('nnoremap <silent> <buffer> <c-n> :execute("b %s")<cr>', responseBufName))
+    execute("b " . responseBufName)
+    nnoremap <silent> <buffer> <c-n> :execute("b " . g:vrc_header_buffer_name)<cr>
```
Comment (diepm) [inline ftplugin/rest.vim]: Try to avoid hidden map.

**Comment by diepm** [comment]
Thanks for this great idea. Please consider my comments below.

* I suggest displaying the headers in a split buffer, probably on top of the output buffer. It's so that everything is clear, and we don't have to worry about the mapping to switch back and forth. This "mini" display would be similar to the quickfix window when we set `vrc_show_command` to 1. The only difference is that we might want this header buffer to be a regular buffer.
* Do you think it makes sense to have the buffer-scoped header buffer name like with the output buffer where a user can set a output buffer name for each "rest" buffer.
* Please update the help file as well (you might want to do that after all code review is done).
* Indentation is 2 spaces.

**Comment by hiberabyss** [comment]
Hi @diepm,

For your first point, we do not care about header content in many cases, showing it in a separate window may be a waste of the space to display response content.

From my experience, the reason I enable curl header is to get the response filetype and have a see of the http status code if there is any error.

How do you think about this?

I will update source code according your comments.

Thanks!

**Comment by diepm** [comment]
We can add two options,

* `vrc_hide_headers`: will hide the headers if they present.
* `vrc_split_headers`: will display the headers part in a split buffer.

Of course these two options only take effect if the response includes headers (i.e., with `-i` option). The "hide" option has a higher priority than the "split" one since if the user wants to hide the header section, we don't have to display it.

What do you think?

**Comment by hiberabyss** [comment]
I think it's a little complicated to add two options for this feature. Maybe we could add one first, and then add another one if there is the demand.

**Pull request closed without merging**